jobs/scheduled/create_holiday_events.rb

```diff
@@ -51,13 +51,23 @@ def execute(args)
           .to_h
 
       # Remove holidays for deactivated/suspended/silenced users
-      CalendarEvent.where(post_id: nil).where.not(user_id: usernames.keys).delete_all
+      CalendarEvent.where(post_id: nil).where.not(user_id: usernames.keys).destroy_all
+
+      # Remove future holidays when users changed their region
+      CalendarEvent
+        .joins(user: :_custom_fields)
+        .where(post_id: nil)
+        .where("start_date > ?", today)
+        .where("user_custom_fields.name = ?", ::DiscourseCalendar::REGION_CUSTOM_FIELD)
+        .where("LENGTH(COALESCE(user_custom_fields.value, '')) > 0")
+        .where("user_custom_fields.value != calendar_events.region")
+        .destroy_all
```

Comment:
This query is the gist of the PR. It will automatically remove **future** holiday events that aren't in the region the users have selected in their profile.